keps/sig-architecture/20190412-conformance-behaviors.md

```diff
@@ -0,0 +1,268 @@
+the payload of the create or update calls, not in the simple fact that those
+endpoints were hit. This means that even if a given endpoint is shown as
+covered, it’s impossible to know how much that tests the actual behavior.
+
+Coverage is measured this way because there is no single, explicit list of
+behaviors that comprise the expected behavior of a conformant cluster. These
+expectations are spread out across the existing design documents, KEPs, the user
+documentation, a subset of the e2e test, and the code itself. This makes it
+impossible to identify if the conformance suite provides a meaningful test of a
+cluster’s operation.
+
+This proposal is to explicitly define a machine-readable list of expected
+behaviors from a conformant cluster. This top-down approach is a lot of work,
+but it has some important benefits:
+
+* The reviewers of “conforming behavior” do not need to be the same people as
+  the reviewers of “this test code validates the behavior”.
+* Distributors or implementors of pluggable components have a single location to
+  reference for expectations.
+* Coverage can be measured by tying specific tests back to the list of
+  behaviors.
+* Tooling can produce an explicit conformance report displaying the behaviors
+  and the status of each test that checks that behavior.
+* Behaviors can be grouped into sets of functionality that are conformant only
+  for specific cluster profiles.
+* It is a forcing function to clearly document each expected conforming
+  behavior.
+
+### Goals
+
+* Enable separate review of behaviors and tests that evaluate those behaviors.
+* Provide a single location for defining conforming behavior.
+* Provide tooling to measure the conformance test coverage of the behaviors.
+* Provide an incremental migration path from current conformance testing to the
+  updated framework.
+* Update the tooling for producing conformance reports.
+
+### Non-Goals
+
+* Develop a complete set of behaviors that define a conforming cluster.
+* Add new conformance tests.
+
+## Proposal
+
+Files that define each behavior in a machine-readable format will
+be created and stored in the k/k repository. These files include identifiers
+for the behaviors that can be used within the e2e tests, so that the test logs
+contain enough information to connect the test back to the behavior.
+
+The structure of the YAML file is described by these Go types:
+
+```
+package conformance
+
+type Area struct {
+	Area      string     `json:"area,omitempty"`
+	Api       string     `json:"api,omitempty"`
+	Behaviors []Behavior `json:"behaviors,omitempty"`
+}
+
+type Behavior struct {
+	Id          string `json:"id,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+```
+
+An `Area` is an arbitrary grouping used to organize tests. The `Api` field is
+optional and may specify the API group to which the tests apply, if any. Each
+behavior has a unique identifier and a description. An example is shown below.
```

Comment:
Do you want to be more prescriptive about structure of `Id`? Seems like there is a common pattern being used in the example below...